### PR TITLE
tests are now functional

### PR DESCRIPTION
**Pull request by savoisn**

### CODE REVIEW

File: generators/index.js [hunk 1]
```diff
@@ -311,7 +310,10 @@ module.exports = generators.Base.extend({
 
     installClient: function(){
       if(!this.options['client-required']) return;
-      this.composeWith('react-loopback:client', {
+
+      var client = require.resolve('../generators/client');
```
Comment (ghamaide) [inline generators/index.js]: const :)

File: generators/index.js [hunk 2]
```diff
@@ -320,7 +322,10 @@ module.exports = generators.Base.extend({
 
     installServer: function(){
       if (!this.options['server-required']) return;
-      this.composeWith('react-loopback:server', {
+
+      var server = require.resolve('../generators/server');
```
Comment (ghamaide) [inline generators/index.js]: const :)

File: test/client-component/index.js [hunk 3]
```diff
@@ -2,24 +2,32 @@ var path = require('path');
 var helpers = require('yeoman-test');
 var assert = require('yeoman-assert');
 var fs = require('fs-extra');
-var generator;
 var _ = require('lodash');
 
+var generator;
```
Comment (ghamaide) [inline test/client-component/index.js]: let

File: test/client-component/index.js [hunk 4]
```diff
@@ -2,24 +2,32 @@ var path = require('path');
...
 
 describe('react-loopback:client-component', function () {
 
+  var tmpFolders = []
```
Comment (ghamaide) [inline test/client-component/index.js]: const

File: test/client-component/index.js [hunk 5]
```diff
@@ -2,24 +2,32 @@ var path = require('path');
...
+
   beforeEach(function(done) {
+    var tmpFolder = path.join(__dirname, _.uniqueId('.tmp-'))
```
Comment (ghamaide) [inline test/client-component/index.js]: const

File: test/crud/index.js [hunk 6]
```diff
@@ -6,19 +6,26 @@ let generator;
...
     .withOptions({})
     .withArguments([]);
     done();
   });
 
   after(function(done) {
-    fs.removeSync(path.join(__dirname, '.tmp*'));
-    done();
+    var promises = []
+    for(let tmpFolder of tmpFolders){
```
Comment (ghamaide) [inline test/crud/index.js]: you could use a functional reduce here :)

File: test/index.js [hunk 7]
```diff
@@ -22,8 +27,13 @@ describe('react-loopback', function () {
...
 
     after(function (done) {
-        fs.removeSync(path.join(__dirname, '.tmp*'));
-        done();
+      var promises = []
+      for(let tmpFolder of tmpFolders){
+          promises.push(fs.remove(tmpFolder));
+      }
+
+      Promise.all(promises).then(() => done());
```
Comment (ghamaide) [inline test/index.js]: Promise.all(promises).then(() => done()); ==> Promise.all(promises).then(done);

File: test/index.js [hunk 8]
```diff
@@ -132,8 +142,8 @@ describe('react-loopback', function () {
             });
     });
 
-    xit('should generate basic package.json file if server and client', function (done) {
-        completePackageJson = fs.readFileSync(path.resolve(__dirname, 'package.client-server.json'), 'utf-8');
+    it('should generate basic package.json file if server and client', function (done) {
+        var completePackageJson = fs.readFileSync(path.resolve(__dirname, 'package.client-server.json'), 'utf-8');
```
Comment (ghamaide) [inline test/index.js]: const

File: test/server-datasource/index.js [hunk 9]
```diff
@@ -8,6 +8,7 @@ var _ = require('lodash');
 
 describe('react-loopback:server-datasource', function () {
 
+  var tmpFolders = []
```
Comment (ghamaide) [inline test/server-datasource/index.js]: const

File: test/index.js (hunk 7 above)
Comment (savoisn) [inline test/index.js]: nearly : Promise.all(promises).then(done());

**Pull request merged**